src/app/editor/_components/createPostForm/CreatePostForm.tsx

```diff
@@ -0,0 +1,313 @@
+"use client";
+import {
+  Button,
+  Chip,
+  Dialog,
+  DialogActions,
+  Grid2 as Grid,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  TextField,
+  Typography,
+  Box,
+  InputLabel,
+  Select,
+  MenuItem,
+  FormHelperText,
+} from "@mui/material";
+import { type CreatePostFormType } from "../../types/types";
+import { Controller, useWatch } from "react-hook-form";
+import { INTEREST_ARRAY } from "~/app/sign-up/_config/config";
+import { useCreatePostForm } from "../../_hooks/useCreatePostForm";
+import { useState } from "react";
+
+export type CreatePostFormProps = {
+  open: boolean;
+  handleClose: () => void;
+  createPostFormData: CreatePostFormType;
+};
+
+export const CreatePostForm: React.FC<CreatePostFormProps> = ({
+  open,
+  handleClose,
+  createPostFormData,
+}) => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useCreatePostForm({
+    defaultValues: createPostFormData,
+  });
+
+  const postType = useWatch({ control, name: "postType" });
```

Comment:
i dont think we need to import useWatch hook here, we can simply get the watch function from useCreatePostForm

Reply:
done